src/App.js

```diff
@@ -11,11 +11,13 @@ import NotFound from "./components/NotFound";
 import ThemeToggler from "./components/ThemeToggler";
 import { Route, Routes } from "react-router-dom";
 import { DataProvider } from "./context/DataContext";
+import VerifyToken from "./services/VerifyToken";
 
 function App() {
   return (
     <div className="App">
       <DataProvider>
+        <VerifyToken />
```

Comment:
I havent seen this way to check for auth before in a React application.
Usually you check auth before going to a route or before doing something critical.
To check on every mouse-action seems a bit overkill. But that is just my opinion.